features/step_definitions/artefact_steps.rb

```diff
@@ -161,3 +172,26 @@
 When /^I try to create a new artefact with the same need$/ do
   visit new_artefact_path(:artefact => {:need_id => @artefact.need_ids.first})
 end
+
+When /^I browse to the "(.*)" URL for the artefact$/ do |path|
+  visit archive_artefact_path(@artefact)
+end
+
+And /^I click the archive tab on the artefact page$/ do
+  visit edit_artefact_path(@artefact)
+  click_link "Archive"
+end
+
+And /^I archive the existing artefact$/ do
+  Artefact.observers.disable :update_search_observer do
+    @artefact.update_attributes!('state' => 'archived')
```

Comment:
The reason I did the `update_attributes!` thing was because I thought it went hand-in-hand with `Artefact.observers.disable ...`. Turns out that keeping `Artefact.observers.disable` to stop errors relating to search not being a registered request in tests, and using `click_button` works. Thanks for making me look again.